src/rust/zh.manhuagui/src/decoder.rs

```diff
@@ -0,0 +1,441 @@
+// use core::{iter::Map, ops::{Index, IndexMut}};
+
+use aidoku::{
+	prelude::format,
+	std::{String, Vec},
+};
+use alloc::vec;
+
+pub struct Decoder {
+	func: String,
+	a: i32,
+	c: i32,
+	data: Vec<String>,
+}
+
+impl Decoder {
+	pub fn new(document: String) -> Self {
+		// aidoku::prelude::println!("document: {}", document);
+
+		let script = get_script(document);
+
+		let func = get_func(script.clone());
+		let a = get_a(script.clone(), func.clone()).parse::<i32>().unwrap();
+		let c = get_c(script.clone(), func.clone()).parse::<i32>().unwrap();
+		let data: Vec<String> = get_data(script, func.clone());
+
+		Decoder { func, a, c, data }
+	}
+
+	fn e(&self, c: i32) -> String {
+		let mut prefix: String = String::new();
+		if c >= self.a {
+			prefix = self.e(c / self.a);
+		}
```

Comment:
try to avoid mutable variables if you can
```suggestion
		let prefix: String = if c >= self.a { self.e(c / self.a) } else { String::new() };
```